fiftyone/core/annotation/attributes.py

```
@@ -0,0 +1,190 @@
+    values: Optional[list] = None
+    when: Optional[list[When]] = None
+
+    def __post_init__(self) -> None:
+        missing = [
+            k for k in ("name", "type", "component") if not getattr(self, k)
+        ]
+        if missing:
+            raise ValueError(
+                f"AttributeSpec missing required field(s): "
+                f"{', '.join(missing)}"
+            )
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)